modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/contexts/Contexts.scala

```diff
@@ -269,16 +266,24 @@ final class Contexts[F[_]](agg: ContextAggregate[F], doms: Domains[F], baseUri:
             }
           }
         case (_, Some(arr), _) =>
-          F.sequence(arr.map(v => expandValue(v)))
+          F.sequence(arr.map(v => resolveValue(v)))
             .map(values =>
               values.foldLeft(Json.obj()) { (acc, e) =>
                 acc deepMerge e
             })
-        case (_, _, Some(_)) => F.pure(value)
+        case (_, _, Some(_)) => F.pure(context)
         case (_, _, _)       => F.raiseError(shapeValidationFailure)
       }
 
-    inner(context).map(json => context deepMerge Json.obj("@context" -> json))
+    def inner(jsonObj: JsonObject): F[JsonObject] =
+      F.sequence(jsonObj.toVector.map {
+        case ("@context", v) => resolveValue(v).map("@context" -> _)
+        case (k, v)          => resolve(v).map(k               -> _)
```

Comment:
This formatting looks a bit weird.

Reply:
It was scalafmt one...but yeah looks a bit weird